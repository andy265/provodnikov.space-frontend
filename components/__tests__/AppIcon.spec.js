const { createRenderer } = require('vue-server-renderer')
const { shallow } = require('vue-test-utils')
const AppIcon = require('~/components/AppIcon')

describe('AppIcon.vue', () => {
  it('has same HTML structure when size is not in props', () => {
    let wrapper = shallow(AppIcon, {
      propsData: {
        type: 'vk'
      }
    })
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('has same HTML structure when size is "2"', () => {
    let wrapper = shallow(AppIcon, {
      propsData: {
        type: 'vk',
        size: 4
      }
    })
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })
})
