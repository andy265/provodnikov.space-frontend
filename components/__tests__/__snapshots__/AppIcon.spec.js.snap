// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIcon.vue has same HTML structure when size is "2" 1`] = `<span aria-hidden="true" class="app-icon fa fa-vk fa-4x"></span>`;

exports[`AppIcon.vue has same HTML structure when size is not in props 1`] = `<span aria-hidden="true" class="app-icon fa fa-vk"></span>`;
